vm/src/dictdatatype.rs

```diff
@@ -248,9 +248,17 @@ impl<T: Clone> Dict<T> {
             if let Some(index) = entry_index.index() {
                 // Update existing key
                 if let Some(entry) = inner.entries.get_mut(index) {
-                    let entry = entry
-                        .as_mut()
-                        .expect("The dict was changed since we did lookup.");
+                    let entry = if let Some(entry) = entry.as_mut() {
+                        entry
+                    } else {
+                        // The dict was changed since we did lookup. Let's try again.
+                        // this is very rare to happen
+                        // (and seems only happen with very high freq gc, and about three to four time in 200000 iters)
+                        // but still possible
+                        // so logging a warn is acceptable in here
+                        warn!("The dict was changed since we did lookup. Let's try again.");
+                        continue;
+                    };
```

Comment:
```suggestion
                    let Some(entry) = entry.as_mut() {
                        // The dict was changed since we did lookup. Let's try again.
                        // this is very rare to happen
                        // (and seems only happen with very high freq gc, and about three to four time in 200000 iters)
                        // but still possible
                        continue;
                    };
```

I don't think a warning is really needed here since we're just going to continue (person receiving warning can't do much about it). Same thing happens in all other cases where something changes the dict.

Do you need the warn for debugging purposes with the gc?

Reply:
Not really, I guess we could remove the warning here, doing it now.